### ISSUE TITLE
Import subscribers with new version broken / Delete subscribers not working

### ISSUE THREAD

**Issue by g4rf** [hizzle-co/noptin#702]
### Description

We use Noptin Newsletter to send e-mails to our members for invitations to meetings about twice to four times a year. To protect our members data we import this data only for sending mails and deleting it afterwards. This workflow worked nice until now. The last mail we send was in June.

When I try to import our CSV now, I got a bunch of errors:
```
Error saving Subscriber: Duplicate entry '0.00000000' for key 'PRIMARY'.
```
and at the end only 1 subscriber is added.
With the old importer I could map the ID field to our ID field, now this is not possible anymore. I think, that's the problem.

Second problem: I tried to delete the one subscriber, but it doesn't work. No error message in console, the HTTP POST gives a `204 No Content` back. But the subscriber is still there.
I looked for the tool **Delete All Subscribers**, but it doesn't exists anymore. It's replaced by a tool **Reset all data**, but I don't want to reset all data – just delete savely all subscribers.

Noptin was a good solution for our usecase. Now it's unuseable anymore, what costs us time to look for another solution. It's bad to break completly a good working thing with an update…

### Screenshots

![grafik](https://github.com/hizzle-co/noptin/assets/18557074/180961bc-b05a-4d83-b355-c0a63def91d7)

### Additional information

Noptin - WordPress Newsletter Plugin 2.0.5
WordPress 6.3.1

Empty theme and no other plugins as we just use this instance for sending the mails.

**Comment by picocodes** [comment]
Hi @g4rf,

Sorry about that. You can click on the "Delete" link found on the subscribers' overview page to delete all subscribers. Do not select any subscriber when doing this.

As for the error when importing subscribers, click on Noptin > Settings > Custom Fields then create a dummy custom field, save the settings, then delete the field and resave your settings again. Let me know if that fixes the issue.

**Comment by g4rf** [comment]
Hi picocodes,

thank you for your fast reply!

I solved it yesterday by deleting completely the plugin and installing version 1. Thereby I had to create the custom fields again.

Tested today the import and now it works like a charme. So I think, there got something broken with the database tables while updating from version 1 to version 2 and then trying the mass import. 
Also the *delete* function is now working as expected, either with selected subscribers and without selecting to delete all.

So let's say, the solution was a complete reinstall, but probably just editing the custom fileds may also solved it. I think if I installed version 2 instead of version 1 the problems may be also gone. But I was in a hurry and so wanted back to a known state. 🥴
